buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/fixtures/TestFixturesPlugin.kt

```diff
@@ -60,46 +55,72 @@ open class TestFixturesPlugin : Plugin<Project> {
         configureAsConsumer()
     }
 
+
+    /**
+     * This mimics what the java-library plugin does, but creating a library of test fixtures instead.
+     */
     private
     fun Project.configureAsProducer() {
+        val testFixtures by java.sourceSets.creating
 
-        configurations {
-            create("outputDirs")
+        java.sourceSets.named<SourceSet>("test") {
+            compileClasspath += testFixtures.output
+            runtimeClasspath += testFixtures.output
+        }
 
-            create("testFixturesCompile") { extendsFrom(configurations["compile"]) }
-            create("testFixturesImplementation") { extendsFrom(configurations["implementation"]) }
-            create("testFixturesRuntime") { extendsFrom(configurations["runtime"]) }
+        val testFixturesJar by tasks.creating(Jar::class) {
+            from(testFixtures.output)
+            classifier = "test-fixtures"
+        }
 
-            // Expose configurations that include the test fixture classes for clients to use
-            create("testFixturesUsageCompile") {
-                extendsFrom(configurations["testFixturesCompile"], configurations["outputDirs"])
+        configurations {
+            val testFixturesCompile by getting {
+                extendsFrom(configurations["compile"])
```

Comment:
❌ Shouldn't we use the constants on `JavaPlugin` instead of plain strings?

Reply:
yep